oneflow/user/ops/ndindex_to_offset_op.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 The OneFlow Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#include "oneflow/core/framework/framework.h"
+namespace oneflow {
+
+namespace user_op {
+
+REGISTER_USER_OP("ndindex_to_offset")
+    .Input("index")
+    .Input("dims")
+    .Output("out")
+    .SetTensorDescInferFn([](user_op::InferContext* ctx) -> Maybe<void> {
+      const TensorDesc* index = ctx->TensorDesc4ArgNameAndIndex("index", 0);
+      const TensorDesc* dims = ctx->TensorDesc4ArgNameAndIndex("dims", 0);
+      int64_t index_num_axes = index->shape().NumAxes();
+      CHECK_GT_OR_RETURN(index_num_axes, 0);
+      int64_t dims_num_axes = dims->shape().NumAxes();
+      CHECK_GT_OR_RETURN(dims_num_axes, 0);
+      CHECK_EQ(index->shape().elem_cnt(), dims->shape().elem_cnt());
+      user_op::TensorDesc* out = ctx->TensorDesc4ArgNameAndIndex("out", 0);
+      *out->mut_shape() = Shape({1});  // Only return single value
```

Comment:
```suggestion
      *out->mut_shape() = Shape({0});  // Only return single value
```
`{1}` 容易让人以为初始值1有什么特殊含义，用0占位即可

Reply:
`// Only return single value` 注释没有必要（也应是 "a single value"），或者改成更明确的 `out tensor only contains a single offset`

Reply:
这里设置为0我哪儿来的空间放offset- -，设0报错了

Reply:
这个是我弄错了，初始化的值是表示形状，还是用 {1}